frontend/src/components/TimerComponent.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import LinearProgress from "@mui/material/LinearProgress";
+import { Box, Button } from "@mui/material";
+
+function TimerComponent({ setText, setTimerCompleted }) {
+  const [progress, setProgress] = useState(100);
+  const [remaining, setRemaining] = useState(20);
+  const [timerId, setTimerId] = useState(null);
+  const [complete, setComplete] = useState(false);
+
+  useEffect(() => {
+    if (complete) {
+      setTimerCompleted(true);
+      setText("Could not find a match");
+    }
+  }, [setTimerCompleted, setText, complete]);
+
+  useEffect(() => {
+    const duration = 20;
+    const interval = 1000;
+
+    const progressTimer = setInterval(() => {
+      setProgress((prevProgress) => {
+        if (prevProgress <= 0) {
+          clearInterval(progressTimer);
+          setComplete(true);
+          return 0;
+        }
+        return prevProgress - (100 / duration) * (interval / 1000);
+      });
+    }, interval);
+
+    const remainingTimer = setInterval(() => {
+      setRemaining((prev) => {
+        if (prev <= 0) {
+          clearInterval(remainingTimer);
+          setComplete(true);
+          return 0;
+        }
+        return prev - 1;
+      });
+    }, interval);
+
+    setTimerId([progressTimer, remainingTimer]);
+
+    return () => {
+      clearInterval(progressTimer);
+      clearInterval(remainingTimer);
+    };
+  }, []);
+
+  const stopTimer = () => {
+    if (timerId != null) {
+      timerId.forEach((id) => clearInterval(id));
+    }
+    setProgress(0);
+    setRemaining(0);
+    setComplete(true);
+    setTimerId([]);
+  };
+
+  return (
+    <div>
+      <LinearProgress variant="determinate" value={progress} />
+      <Box display="flex" justifyContent="space-between" width="100%" mt={1}>
+        <div>Remaining Time: {remaining} seconds</div>
```

Comment:
I added a component to reflect time remaining for the matching & button to cancel the matching, which may be more intuitive from user perspective. However, need to check with backend on how to cancel the matching request before 20 seconds is reached